### PR TITLE
Improved cross-validation

### PR DESCRIPTION
**Pull request by ahwillia**
See `crossval-demo.ipynb` notebook

### CODE REVIEW

File: twpca/crossval.py [hunk 1]
```diff
@@ -11,94 +11,52 @@
...
-        yield train_idx, test_idx
-
-def leave_k_out_iter(N, K):
-    """Leave K out cross validation iterator
+    Args
+    ---
+    model : TWPCA model instance
+    data (ndarray) : data tensor, trials x time x neurons
+    nfits (int) : number of cross-validation runs
+    censor_prob (float) : probability of setting element of data to nan
```
Comment (poolio) [inline twpca/crossval.py]: why censor_prob? could we stick with keep_prob or drop_prob like folks use for dropout?

also make the doc more descriptive, like "fraction of data to hold out for each cross validation run"

File: twpca/crossval.py [hunk 2]
```diff
@@ -11,94 +11,52 @@
...
 
-    Args:
-        N: int, number of datapoitns
-        K: int, number to leave out in training set
+    Note: additional keyword args are passed to model.fit(...)
```
Comment (poolio) [inline twpca/crossval.py]: please add docs on what this returns

File: twpca/crossval.py [hunk 3]
```diff
@@ -148,12 +100,7 @@ def hyperparam_search(data, n_components, warp_scales, time_scales,
         time_reg: function, takes scalar and outputs a regularization term (in tensorflow) for
                     the time factors. By default,
                     `time_reg = twpca.regularizers.curvature(s, power=2, axis=0)`.
-        crossval_method: str, specifies cross validation. One of {'kfold' (default), 'leavout'}.
-        K: int, cross validation parameter. For example, K = 5 specifies 5-fold cross validation
-                    when crossval_method = 'kfold', and K = 1 specifies leave-1-out validation
-                    when crossval_method = 'leaveout'
-        max_fits: int, number of fits per model. For example max_fits = 1 means that only a single
-                  training and test set are evaluated (default: np.inf).
+        nfits: int, number of crossvalidation runs per model.
```
Comment (poolio) [inline twpca/crossval.py]: doc other options?

File: twpca/utils.py [hunk 4]
```diff
@@ -103,3 +103,13 @@ def compute_lowrank_factors(data, n_components, fit_trial_factors, nonneg, last_
...
 
+
+def correlate_nanmean(x, y, **kwargs):
+    """Wrapper around np.correlate that handles NaNs."""
+    x_ = np.nan_to_num(x)
+    y_ = np.nan_to_num(y)
+    xcorr = np.correlate(x_, y_, **kwargs)
+    xcorr_count = np.correlate((~np.isnan(x)).astype(float),
+                               (~np.isnan(y)).astype(float), mode='same')
+    xcorr_avg = xcorr / xcorr_count
```
Comment (poolio) [inline twpca/utils.py]: this normalization doesn't work and biases initialization to the boundaries. I'll ad a commit to remove it

**Review by poolio — COMMENTED** [review]
Main complaint is just the name censor_prob, but otherwise looks good. From the notebook it looks like rank 2 is still not what you'd choose from CV. Any idea if this is just noise or if we need to be more selective with how we drop out elements of the tensor?

File: twpca/crossval.py (hunk 1 above)
Comment (ahwillia) [reply twpca/crossval.py]: I like `drop_prob` but originally decided against it since dropout doesn't seem like a great analogy for what we're doing... But I'll switch to that.
